chrome-extension/content.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Content script for all urls 
+ */
+
+//Insert bulma stylesheet
+//TODO: STOP USING BULMA ONCE I WRITE MY OWN CSS FOR ALL EXTENSION ELEMENTS
+const bulmaLink = document.createElement('link');
+bulmaLink.setAttribute('href','https://cdn.jsdelivr.net/npm/bulma@0.9.0/css/bulma.min.css');
+bulmaLink.setAttribute('rel','stylesheet');
+document.head.appendChild(bulmaLink)
+
+//Inject invisible sidebar onto page
+const sidebar = document.createElement('div');
+sidebar.id = 'lobstericecream';
+sidebar.classList.add('panel','injection-panel','px-4','py-4');
+document.body.appendChild(sidebar); 
+
+//Inject invisible highlight icon onto page
+const iconButton = document.createElement('button');
+iconButton.classList.add('injection-icon');
+iconButton.innerHTML = 'Save';
+//TODO: CHANGE BUTTON TO AN ACTUAL ICON!! 
+
+document.body.appendChild(iconButton);
+
+//Keep track of last used highlight
+let lastContent; 
+
+//Show extension icon when selecting text on page
+document.onclick = (event) => {
+  const dontListen = '.injection-panel, .injection-icon';
+  if (iconButton.style.display === 'block') {
+    iconButton.style.display ='none'; 
+  } 
+  const content = window.getSelection().toString();
+  if (content && content.length > 0 && content !== lastContent) {
+    showIcon(event, content);
+    lastContent = content;
+    chrome.storage.sync.set({ content });
```

Comment:
I admit that I don't know much about how the Chrome storage works. Why is it necessary to save the content?

Reply:
This was originally for accesssing selected text in popup bar, but that is no longer what we're doing so i can get rid of it.